application/extensions/AdvancedSettingWidget/views/float.php

```diff
@@ -0,0 +1,23 @@
+<div class="input-group col-12">
+    <?php if (isset($this->setting['aFormElementOptions']['inputGroup']['prefix'])) : ?>
+        <div class="input-group-addon">
+            <?= $this->setting['aFormElementOptions']['inputGroup']['prefix']; ?>
+        </div>
+    <?php endif; ?>
+    <input
+        type="number" 
+        class="form-control" 
+        name="<?= $inputBaseName ?>"
+        id="<?= CHtml::getIdByName($inputBaseName); ?>"
+        <?= ($this->setting['help']) ? 'aria-describedby="help-' . CHtml::getIdByName($inputBaseName) . '"' : "" ?>
+        value="<?= CHtml::encode($this->setting['value']); ?>"
+        min="<?= $this->setting['min'] ?? ''?>"
+        max="<?= $this->setting['max'] ?? ''?>"
+        pattern="^[-+]?[0-9]*\.[0-9]+$"
```

Comment:
Why ? Type number do the restriction.

Pattern only need for integer or specific number